sdk/src/main/scala/com/horizen/account/AccountSidechainNodeViewHolder.scala

```diff
@@ -0,0 +1,281 @@
+    case msg: AbstractSidechainNodeViewHolder.ReceivableMessages.GetDataFromCurrentSidechainNodeView[
+      AccountTransaction[Proposition, Proof[Proposition]],
+      AccountBlockHeader,
+      AccountBlock,
+      AccountFeePaymentsInfo,
+      NodeAccountHistory,
+      NodeAccountState,
+      NodeWalletBase,
+      NodeAccountMemoryPool,
+      AccountNodeView,
+      _] @unchecked =>
+      msg match {
+        case AbstractSidechainNodeViewHolder.ReceivableMessages.GetDataFromCurrentSidechainNodeView(f) => try {
+          val l: AccountNodeView = new AccountNodeView(history(), minimalState(), vault(), memoryPool())
+          sender() ! f(l)
+        }
+        catch {
+          case e: Exception => sender() ! akka.actor.Status.Failure(e)
+        }
+
+      }
+  }
+
+  override protected def applyFunctionOnNodeView: Receive = {
+    case msg: AbstractSidechainNodeViewHolder.ReceivableMessages.ApplyFunctionOnNodeView[
+      AccountTransaction[Proposition, Proof[Proposition]],
+      AccountBlockHeader,
+      AccountBlock,
+      AccountFeePaymentsInfo,
+      NodeAccountHistory,
+      NodeAccountState,
+      NodeWalletBase,
+      NodeAccountMemoryPool,
+      AccountNodeView,
+      _] @unchecked =>
+      msg match {
+        case AbstractSidechainNodeViewHolder.ReceivableMessages.ApplyFunctionOnNodeView(f) => try {
+          val l: AccountNodeView = new AccountNodeView(history(), minimalState(), vault(), memoryPool())
+          sender() ! f(l)
+        }
+        catch {
+          case e: Exception => sender() ! akka.actor.Status.Failure(e)
+        }
+
+      }
+
+  }
+
+  override protected def applyBiFunctionOnNodeView[T, A]: Receive = {
+    case msg: AbstractSidechainNodeViewHolder.ReceivableMessages.ApplyBiFunctionOnNodeView[
+      AccountTransaction[Proposition, Proof[Proposition]],
+      AccountBlockHeader,
+      AccountBlock,
+      AccountFeePaymentsInfo,
+      NodeAccountHistory,
+      NodeAccountState,
+      NodeWalletBase,
+      NodeAccountMemoryPool,
+      AccountNodeView,
+      T,A] @unchecked =>
+      msg match {
+        case AbstractSidechainNodeViewHolder.ReceivableMessages.ApplyBiFunctionOnNodeView(f,functionParams) => try {
+          val l: AccountNodeView = new AccountNodeView(history(), minimalState(), vault(), memoryPool())
+          sender() ! f(l,functionParams)
+        }
+        catch {
+          case e: Exception => sender() ! akka.actor.Status.Failure(e)
+        }
+
+      }
+  }
+
+  // TODO FOR MERGE
+  override val listOfStorageInfo: Seq[SidechainStorageInfo] = Seq()
+
+
+  // TODO FOR MERGE: implement these methods
+  override def dumpStorages(): Unit = ???
+  override def getStorageVersions: Map[String, String] = ???
+
+  override def processLocallyGeneratedTransaction: Receive = {
+    case newTxs: LocallyGeneratedTransaction[SidechainTypes#SCAT] =>
+      newTxs.txs.foreach(tx => {
+        // TODO FOR MERGE - Any custom implementation?
```

Comment:
let's keep it.
Moreover we are going to have this method common to both models. In account model we are going to introduce the same check but with different context (gas instead of fee)